dffml/util/net.py

```diff
@@ -432,26 +287,20 @@ def on_error(func, path, exc_info):
     directory_path = pathlib.Path(directory_path)
 
     async def extractor(download_path):
-        download_path = download_path.absolute()
-        with chdir(directory_path):
-            try:
-                shutil.unpack_archive(str(download_path), ".")
-            except Exception as error:
-                shutil.rmtree(directory_path, onerror=on_error)
-                raise DirectoryNotExtractedError(directory_path) from error
-
-    extract = cached_download(
-        url, file_path, expected_hash, protocol_allowlist=protocol_allowlist
-    )(extractor)
-
-    def mkwrapper(func):
-        @functools.wraps(func)
-        async def wrapper(*args, **kwds):
-            if not directory_path.is_dir():
-                directory_path.mkdir(parents=True)
-                await extract()
-            return await func(*(list(args) + [directory_path]), **kwds)
-
-        return wrapper
-
-    return mkwrapper
+        try:
+            shutil.unpack_archive(str(download_path), str(directory_path))
+        except Exception as error:
+            shutil.rmtree(directory_path, onerror=on_error)
+            raise DirectoryNotExtractedError(directory_path) from error
```

Comment:
took me solid hours to figure this one out.  Apparently ` shutil.unpack_archive()` doesn't recognize "__.zip" as a zip file when using `with chdir(directory_path):`, I suppose we're returning the path to the zip file directly instead of the directory containing it so that could be causing it, so I just passed the path directly to ` shutil.unpack_archive()`.

Reply:
The test was weirdly passing when I didn't use Asynio.run() so it had me running in circles around using loops and `asyncio.run()`.